relay-server/src/actors/processor.rs

```diff
@@ -594,7 +594,7 @@ pub struct ProjectMetrics {
 /// Encodes metrics into an envelope ready to be sent upstream.
 #[derive(Debug)]
 pub struct EncodeMetrics {
-    pub scopes: HashMap<Scoping, ProjectMetrics>,
+    pub scopes: Vec<(Scoping, ProjectMetrics)>,
```

Comment:
What's the reason for this change?

Reply:
guaranteeing consistency when testing, the test case depends on which scoping->projectmetrics pair come first, but the hashmap doesn't guarantee order.

Reply:
Could we validate that the elements we're looking for exist in the hashmap instead? Assuming this happens with the newly added test `test_ratelimit_per_batch`.

Reply:
Does it change anything, e.g. performance by changing the underlying container `Vec` vs. `HashMap`?  Or anything else between those lines?

Reply:
could do it, but then the test wouldn't consistently fail without this change though, not sure how important that is

Reply:
I'd stick to the hashmap if possible. I don't think the test should impact the data structure in this case

Reply:
how would I verify it works if i use a hashmap?

Reply:
Verify that the map contains all the items you expect, and nothing more. For example, if you expect `{'a': 1, 'b': 2}`, assert that the map contains exactly 2 items and that those items are `'a': 1` and `'b': 2`.

Reply:
without this PR the order of how the hashmap iterates over the keys matters in prod, as it'll short-circuit on the first is_rate_limited == true, so to demonstrate that it's no longer the case with 'continue' i changed it to btreemap. In general we also even prefer btreemap on small maps like this